### ISSUE TITLE
Unable to commit on Windows

### ISSUE THREAD

**Issue by clayboone** [NeogitOrg/neogit#350]
### Description

With a minimal config, the commit window fails to appear with error:

```text
128 git --no-optional-locks commit (0.018 ms) [stderr 12]
  | 
  | *** Please tell me who you are.
  ...
```

My git config:
```cmd
C:\Windows\System32>git config --global user.name
Clay Boone

C:\Windows\System32>git config --global user.email
<my_correct_email@address.com>
```

### Neovim version

NVIM v0.7.2
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compiled by runneradmin@fv-az276-503

Features: -acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files/nvim/share/nvim"

Run :checkhealth for more info

### Operating system and version

Microsoft Windows [Version 10.0.19044.2006]

### Steps to reproduce

1. Install GitBash.
2. Open cmd.exe and configure git user.name and user.email.
3. Delete all nvim configuration and data.
4. Copy the minimal configuration below to `%LocalAppData%/nvim/init.lua`.
5. Launch nvim-qt.exe.
6. `:cd` to a git repo and make some change.
7. Launch Neogit with `:Neogit`.
8. Stage your change.
9. Press `c` and then `c` again.

### Expected behavior

The commit message editor appears.

### Actual behavior

This error appears:

```text
128 git --no-optional-locks commit (0.018 ms) [stderr 12]
  | 
  | *** Please tell me who you are.
  | 
  | Run
  | 
  |   git config --global user.email "you@example.com"
  |   git config --global user.name "Your Name"
  | 
  | to set your account's default identity.
  | Omit --global to set the identity only in this repository.
  | 
  | fatal: unable to auto-detect email address (got 'Boone@DESKTOP-TB4RSSU.(none)')
```

### Minimal config

```Lua
local ensure_packer = function() -- {{{1
  local fn = vim.fn
  local install_path = fn.stdpath("data") .. "/site/pack/packer/start/packer.nvim"
  if fn.empty(fn.glob(install_path)) > 0 then
    fn.system({"git", "clone", "--depth", "1", "https://github.com/wbthomason/packer.nvim", install_path})
    vim.cmd [[packadd packer.nvim]]
    return true
  end
  return false
end

local packer_bootstrap = ensure_packer() -- 1}}}

require("packer").startup(function(use)
  use "wbthomason/packer.nvim"

  use "nvim-lua/plenary.nvim"
  use "TimUntersberger/neogit"

  if packer_bootstrap then require("packer").sync() end
end)

require "neogit".setup{}

vim.api.nvim_create_autocmd('BufWritePost', {
  group = vim.api.nvim_create_augroup('PACKER', { clear = true }),
  pattern = 'init.lua',
  command = 'source <afile> | PackerSync'
}) -- 1}}}
-- vim: fdm=marker
```

**Comment by cwebster2** [comment]
I am reproducing this issue on Linux.  Commit 64245bb7f577bad0308d77dc1116ce7d8428f27f works, though I'm not sure what commit after that is the breaking change.

**Comment by cwebster2** [comment]
Adding to this:

The commit 8adf22f103250864171f7eb087046db8ad296f78 breaks commit with an error, and the next commit 9c63259c56e13e3385705fd44722f31381045c0c fixes the error but breaks committing as seen in this bug.

**Comment by derekthecool** [comment]
I can confirm this error on windows using commit `2a71a5595b49da8a21a20ab8644df9ad0b856ebb`. 
I've noticed this problem for at least a few months.
Though I have a different error message.

```
  1 git --no-optional-locks commit (0.502 ms) [stderr 3]
  | C:\Users\Derek Lomax\scoop\apps\neovim-nightly\current\bin\nvim.exe --headless --clean --noplugin -n -R -c "set runtimepath^=C:\Users\Derek\ Lomax\AppData\Local\nvim-data\site\pack\packer\start\neogit" -c "lua require(""neogit.client"").client()": C:UsersDerek: command not found
  | error: There was a problem with the editor 'C:\Users\Derek Lomax\scoop\apps\neovim-nightly\current\bin\nvim.exe --headless --clean --noplugin -n -R -c "set runtimepath^=C:\Users\Derek\ Lomax\AppData\Local\nvim-data\site\pack\packer\start\neogit" -c "lua require(""neogit.client"").client()"'.
  | Please supply the message using either -m or -F option.
```

I am also using wsl Ubuntu Linux with the same version without commit errors.

**Comment by ten3roberts** [comment]
The above error seems to be due to the git envs not properly merging. The fix for that is in this PR #358
